File: libs/adsp-service-spring-sdk/src/main/java/ca/ab/gov/alberta/adsp/sdk/registration/ConfigurationUpdate.java
package ca.ab.gov.alberta.adsp.sdk.registration;

import com.fasterxml.jackson.annotation.JsonProperty;

class ConfigurationUpdate<T> {
  @JsonProperty
  private final String operation = "UPDATE";
  @JsonProperty
  private final T update;

  public String getOperation() {
    return operation;
  }

  public T getUpdate() {
    return update;
  }

  public ConfigurationUpdate(T update) {
    this.update = update;
  }
}

File: libs/adsp-service-net-sdk/Registration/ConfigurationUpdate.cs

using System.Text.Json.Serialization;

namespace Adsp.Sdk.Registration;

internal sealed class ConfigurationUpdate<T>
{
  [JsonPropertyName("operation")]
  public string Operation { get; } = "UPDATE";


  [JsonPropertyName("update")]
  public IDictionary<string, T> Update { get; set; } = new Dictionary<string, T>();
}
